pwchem/protocols/VirtualDrugScreening/protocol_ligand_characterization.py

```diff
@@ -0,0 +1,144 @@
+                    label='Calculate constitutional descriptors')
+        form.addParam('useElectronic', params.BooleanParam, default=True,
+                    label='Calculate electronic descriptors')
+        form.addParam('useTopological', params.BooleanParam, default=True,
+                    label='Calculate topological descriptors')
+        form.addParam('useGeometrical', params.BooleanParam, default=True,
+                    label='Calculate geometrical descriptors')
+        form.addParam('useRing', params.BooleanParam, default=True,
+                    label='Calculate ring descriptors')
+        form.addParam('useFragment', params.BooleanParam, default=True,
+                    label='Calculate fragment descriptors')
+        form.addParam('useOther', params.BooleanParam, default=True,
+                    label='Calculate other descriptors')
+
+    def _insertAllSteps(self):
+        self._insertFunctionStep('runDescriptorCalc')
+        self._insertFunctionStep('createOutputStep')
+
+    def runDescriptorCalc(self):
+        mol_set = self.inputSmallMolecules.get()
+
+        # Get all available RDKit descriptors
+        descriptor_list = Descriptors.descList
+        self.header = [name for name, _ in descriptor_list]
+        self.propertyDict={}
+
+        self. descriptor_categories = {'other': ['MaxAbsEStateIndex', 'MaxEStateIndex', 'MinAbsEStateIndex', 'MinEStateIndex', 'qed', 'SPS', 'MaxPartialCharge', 'MinPartialCharge', 'MaxAbsPartialCharge', 'MinAbsPartialCharge', 'FpDensityMorgan1', 'FpDensityMorgan2', 'FpDensityMorgan3', 'BCUT2D_MWHI', 'BCUT2D_MWLOW', 'BCUT2D_CHGHI', 'BCUT2D_CHGLO', 'BCUT2D_LOGPHI', 'BCUT2D_LOGPLOW', 'BCUT2D_MRHI', 'BCUT2D_MRLOW', 'AvgIpc', 'BalabanJ', 'BertzCT', 'Ipc', 'PEOE_VSA1', 'PEOE_VSA10', 'PEOE_VSA11', 'PEOE_VSA12', 'PEOE_VSA13', 'PEOE_VSA14', 'PEOE_VSA2', 'PEOE_VSA3', 'PEOE_VSA4', 'PEOE_VSA5', 'PEOE_VSA6', 'PEOE_VSA7', 'PEOE_VSA8', 'PEOE_VSA9', 'SMR_VSA1', 'SMR_VSA10', 'SMR_VSA2', 'SMR_VSA3', 'SMR_VSA4', 'SMR_VSA5', 'SMR_VSA6', 'SMR_VSA7', 'SMR_VSA8', 'SMR_VSA9', 'SlogP_VSA1', 'SlogP_VSA10', 'SlogP_VSA11', 'SlogP_VSA12', 'SlogP_VSA2', 'SlogP_VSA3', 'SlogP_VSA4', 'SlogP_VSA5', 'SlogP_VSA6', 'SlogP_VSA7', 'SlogP_VSA8', 'SlogP_VSA9', 'EState_VSA1', 'EState_VSA10', 'EState_VSA11', 'EState_VSA2', 'EState_VSA3', 'EState_VSA4', 'EState_VSA5', 'EState_VSA6', 'EState_VSA7', 'EState_VSA8', 'EState_VSA9', 'VSA_EState1', 'VSA_EState10', 'VSA_EState2', 'VSA_EState3', 'VSA_EState4', 'VSA_EState5', 'VSA_EState6', 'VSA_EState7', 'VSA_EState8', 'VSA_EState9', 'FractionCSP3', 'NHOHCount', 'NOCount', 'NumAliphaticCarbocycles', 'NumAliphaticHeterocycles', 'NumAmideBonds', 'NumAromaticCarbocycles', 'NumAromaticHeterocycles', 'NumAtomStereoCenters', 'NumBridgeheadAtoms', 'NumHeteroatoms', 'NumHeterocycles', 'NumRotatableBonds', 'NumSaturatedCarbocycles', 'NumSaturatedHeterocycles', 'NumSaturatedRings', 'NumSpiroAtoms', 'NumUnspecifiedAtomStereoCenters', 'Phi'], 
```

Comment:
This big dictionaries should be better placed in the constants.py file and imported where needed.